src/main/java/com/iota/iri/service/milestone/impl/LatestSolidMilestoneTrackerImpl.java

```diff
@@ -94,6 +96,87 @@
      */
     private TransactionRequester transactionRequester;
 
+    /**
+     * Holds information about the current synchronisation progress.
+     */
+    private SyncProgressInfo syncProgressInfo = new SyncProgressInfo();
+
+    /**
+     * Holds variables containing information needed for sync progress calculation.
+     */
+    private static class SyncProgressInfo {
+        /**
+         * The actual start milestone index from which the node started from when syncing up.
+         */
+        private int syncMilestoneStartIndex;
+
+        /**
+         * Used to calculate the average time needed to apply a milestone.
+         */
+        private ArrayBlockingQueue<Long> lastMilestoneApplyTimes = new ArrayBlockingQueue<>(100);
```

Comment:
While you are cutting and pasting you can change the declarative type to `BlockingQueue` if you want.
We have the [LooseCoupling](http://www.ing.iac.es/~docs/external/java/pmd/xref/net/sourceforge/pmd/rules/design/LooseCoupling.html) configured in PMD but apparently it doesn't catch this...

So technically it is something we enforce, but since it can't be automated (I believe annoying rules should be automated) and it is a private member in a private class then I won't be insistent on it. Just try to abstract what you can in general.

Reply:
Changed